programmers/체육복.js

```diff
@@ -0,0 +1,26 @@
+function solution(n, lost, reserve) {
+    let haveNum = n - lost.length; 
+
+    for (let i = 0; i < lost.length; i++) {
+        if (reserve.indexOf(lost[i]) !== -1) {
+            reserve.splice(reserve.indexOf(lost[i]), 1);
+            lost.splice(i, 1);
+            haveNum ++;
+            i--;				
+        }
+    }
+
+    for (let num of lost) {
+        let frontCheck = reserve.indexOf(num - 1)
+        let behindCheck = reserve.indexOf(num + 1)
+
+        if (frontCheck !== -1) {
+            reserve.splice(frontCheck, 1);
+            haveNum++;
+        } else if (behindCheck !== -1) {
+            reserve.splice(behindCheck, 1);
+            haveNum++;
+        }
+    }
+    return haveNum;
+}
```

Comment:
반복문 안에서 변수 선언은 지양할 것
그리고 세미콜론(;) 빼먹지 말 것
```js
  let [frontCheck, behindCheck] = [0, 0];
```
이건 자주 보여준 방식인데 구조분해할당 방식으로 비슷한 변수끼리 모아서 선언한 것임
```suggestion
function solution(n, lost, reserve) {
    let [frontCheck, behindCheck] = [0, 0];
    let haveNum = n - lost.length; 

    for (let i = 0; i < lost.length; i++) {
        if (reserve.indexOf(lost[i]) !== -1) {
            reserve.splice(reserve.indexOf(lost[i]), 1);
            lost.splice(i, 1);
            haveNum ++;
            i--;				
        }
    }

    for (let num of lost) {
        frontCheck = reserve.indexOf(num - 1);
        behindCheck = reserve.indexOf(num + 1);

        if (frontCheck !== -1) {
            reserve.splice(frontCheck, 1);
            haveNum++;
        } else if (behindCheck !== -1) {
            reserve.splice(behindCheck, 1);
            haveNum++;
        }
    }
    return haveNum;
}
```

Reply:
이런식으로 풀 수도 있음
```js
function solution(n, lost, reserve) {
    const setWorkoutClothes = [];
    let answer = 0;
    
    for (let i = 0; i < n; i++) {
        setWorkoutClothes.push(1);
    }
    
    for (let i = 0; i < lost.length; i++) {
        setWorkoutClothes[lost[i] - 1]--;
    }
    
    for (let i = 0; i < reserve.length; i++) {
        setWorkoutClothes[reserve[i] - 1]++;
    }

    for (let i = 0; i < n; i++) {
        if (setWorkoutClothes[i] === 0) {
            if (setWorkoutClothes[i - 1] === 2) {
                setWorkoutClothes[i]++;
                setWorkoutClothes[i - 1]--;
            }
            else if (setWorkoutClothes[i + 1] === 2) {
                setWorkoutClothes[i]++;
                setWorkoutClothes[i + 1]--;
            }
        }
        if (setWorkoutClothes[i] >= 1) answer++;
    }
    return answer;
}
```

n만큼 1을 setWorkoutClothes에 push하고 
```js
    for (let i = 0; i < n; i++) {
        setWorkoutClothes.push(1);
    }
```
setWorkoutClothes의 lost[i] - 1에 위치한 값을 `--` 함
그리고 다음으로 setWorkoutClothes의 reserve - 1에 위치한 값을 `++`함
```js
    for (let i = 0; i < lost.length; i++) {
        setWorkoutClothes[lost[i] - 1]--;
    }
    
    for (let i = 0; i < reserve.length; i++) {
        setWorkoutClothes[reserve[i] - 1]++;
    }
```
마지막으로 setWorkoutClothes[i]의 값이 0일때 양 옆의 요소를 확인함
요소의 값이 2라면 여벌의 체육복이 있다는 뜻이기 때문에 i는 `++`시키고 i - 1 또는 i + 1의 값은 `--`함
```js
    for (let i = 0; i < n; i++) {
        if (setWorkoutClothes[i] === 0) {
            if (setWorkoutClothes[i - 1] === 2) {
                setWorkoutClothes[i]++;
                setWorkoutClothes[i - 1]--;
            }
            else if (setWorkoutClothes[i + 1] === 2) {
                setWorkoutClothes[i]++;
                setWorkoutClothes[i + 1]--;
            }
        }
        if (setWorkoutClothes[i] >= 1) answer++;
    }
    return answer;
```